.circleci/config.yml

```diff
@@ -13,7 +12,38 @@ jobs:
           command: yarn test
           name: Run tests via yarn
 
+jobs:
+  test:
+    executor:
+      name: node/default
+    steps:
+      - test_steps
+
+  test_build_and_publish:
+    executor:
+      name: node/default
+    steps:
+      - test_steps
+      - run:
+          command: yarn build
+          name: Compile TS to JS
+      - run:
+          name: Authenticate with registry
+          command: echo "//registry.npmjs.org/:_authToken=$npm_TOKEN" > ~/hub-react/.npmrc
+      - run:
+          name: Publish package
+          command: npm publish --dry-run
+
 workflows:
   test_my_app:
     jobs:
       - test
+
+  publish:
+    jobs:
+      - test_build_and_publish:
+          filters:
+            tags:
+              only: /^v.*/
+            branches:
+              ignore: /.*/
```

Comment:
The `publish` CI workflow will (supposedly) run when we publish a git tag that contains a version string.